repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java

```diff
@@ -0,0 +1,75 @@
+package org.apache.atlas.tasks;
+
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.model.tasks.AtlasTask;
+import org.apache.atlas.model.tasks.TaskSearchParams;
+import org.apache.atlas.model.tasks.TaskSearchResult;
+import org.apache.atlas.repository.Constants;
+import org.apache.atlas.repository.graphdb.AtlasGraph;
+import org.apache.atlas.repository.graphdb.AtlasIndexQuery;
+import org.apache.atlas.repository.graphdb.AtlasVertex;
+import org.apache.atlas.repository.graphdb.DirectIndexQueryResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.apache.atlas.tasks.TaskRegistry.toAtlasTask;
+
+@Component
+public class AtlasTaskService implements TaskService {
+    private static final Logger LOG = LoggerFactory.getLogger(AtlasTaskService.class);
+
+    private final AtlasGraph graph;
+
+    @Inject
+    public AtlasTaskService(AtlasGraph graph) {
+        this.graph = graph;
+    }
+
+    @Override
+    public TaskSearchResult getTasks(TaskSearchParams searchParams) throws AtlasBaseException {
+        TaskSearchResult ret = new TaskSearchResult();
+        List<AtlasTask> tasks = new ArrayList<>();
+        AtlasIndexQuery indexQuery = null;
+        DirectIndexQueryResult indexQueryResult;
+
+        try {
```

Comment:
Can we use one try-catch instead of one inside the other?

Reply:
I think It is fine if we nest try-catch, this is used here so that a specific exception can be caught here while outer try-catch can deal with other exceptions along with catching the inner one

Reply:
done